### ISSUE TITLE
Milestone 1 Review

### ISSUE THREAD

**Issue by Ivyqiuhan** [UBC-MDS/census-income-prediction#56]
Nice job! I provide here some comments and your grades for the first milestone. Please address these concerns in your third milestone submission.

1. **Draft a Team work contract: Correctness** 

2. **Project set-up: Mechanics**

- Fixing typos:  so long as the changes are made in the source file.

3. **Project proposal: reasoning**

- What about data visualization? What specifically are you going to do? Will you do a heat map with correlation to make sure that variables are not redundant? 
- What about class balance, there could be an imbalance in the classes, in which you would have to under sample or oversample. 
- What about missing data, how will you handle the missing data?
- Logistic regression is a regression model, not a classification model. 
- For these algorithms, what packages will you use? Have you thought of using wrapper algorithms (boruta algorithm) for feature selection?
- A suggestion for metrics, to determine the performance of your models is Area Under Curve (AUC). The Area Under the Curve (AUC) is the measure of the ability of a classifier to distinguish between classes and is used as a summary of the ROC curve. Also look into and SHAP (SHapley Additive exPlanations) which explains the direction of each variable compared to the outcome variable. 

4. **A script that downloads the data: Accuracy**

5. **A script that downloads the data: Quality**

6. **Exploratory data analysis in a literate code document: QUALITY**

- "We may deal with this problem when we build the pipeline and model after we see significant drawback." You should definately balance the sample. I suggest generating synthetic data for oversampling. 

7. **Exploratory data analysis in a literate code document: VIZ**

- If figure captions are not provided the plot should be clearly explained in the text. I would recommend using figure captions.

8. **Exploratory data analysis in a literate code document: REASONING**

- maybe to change "we will deal with class imbalance" instead of "we may deal with...". 

9. **Exploratory data analysis in a literate code document: ACCURACY**
10. **Expectations: Mechanics**

**Comment by nd265** [comment]
> Nice job! I provide here some comments and your grades for the first milestone. Please address these concerns in your third milestone submission.
> 
> 1. **Draft a Team work contract: Correctness**
> 2. **Project set-up: Mechanics**
> 
> * Fixing typos:  so long as the changes are made in the source file.
> 
> 3. **Project proposal: reasoning**
> 
> * What about data visualization? What specifically are you going to do? Will you do a heat map with correlation to make sure that variables are not redundant?
> * What about class balance, there could be an imbalance in the classes, in which you would have to under sample or oversample.
> * What about missing data, how will you handle the missing data?
> * Logistic regression is a regression model, not a classification model.
> * For these algorithms, what packages will you use? Have you thought of using wrapper algorithms (boruta algorithm) for feature selection?
> * A suggestion for metrics, to determine the performance of your models is Area Under Curve (AUC). The Area Under the Curve (AUC) is the measure of the ability of a classifier to distinguish between classes and is used as a summary of the ROC curve. Also look into and SHAP (SHapley Additive exPlanations) which explains the direction of each variable compared to the outcome variable.
> 
> 4. **A script that downloads the data: Accuracy**
> 5. **A script that downloads the data: Quality**
> 6. **Exploratory data analysis in a literate code document: QUALITY**
> 
> * "We may deal with this problem when we build the pipeline and model after we see significant drawback." You should definately balance the sample. I suggest generating synthetic data for oversampling.
> 
> 7. **Exploratory data analysis in a literate code document: VIZ**
> 
> * If figure captions are not provided the plot should be clearly explained in the text. I would recommend using figure captions.
> 
> 8. **Exploratory data analysis in a literate code document: REASONING**
> 
> * maybe to change "we will deal with class imbalance" instead of "we may deal with...".
> 
> 9. **Exploratory data analysis in a literate code document: ACCURACY**
> 10. **Expectations: Mechanics**

Hi @Ivyqiuhan, we cannot see the grades here with the comments, is there any other place where we might see them? 
Also, thanks for your feedback, we will work on it. Some of the things that you have mentioned, we have incorporated in milestone 2, so those things we can just skip for milestone 3 right? 
Thanks

**Comment by PhilsChan** [comment]
Hi @Ivyqiuhan I have another question.
Regarding your comments on our project proposal and EDA, do we have to improve these documents in `proposal.md` and `EDA.ipynb`? Do we also have to reflect our update on our pipeline & final report?

**Comment by Ivyqiuhan** [comment]
If you already fixed those issues in M2, you can skip it for M3 : )

grade should be released on Canvas very soon

**Comment by Ivyqiuhan** [comment]
> Hi @Ivyqiuhan I have another question. Regarding your comments on our project proposal and EDA, do we have to improve these documents in `proposal.md` and `EDA.ipynb`? Do we also have to reflect our update on our pipeline & final report?

please try to enhance your proposaL, EDA, pipeline and final report with my review (you can decide how to address those issues within your project), keep in mind that we are grading the project as a whole

**Comment by nd265** [comment]
Hey, @Ivyqiuhan , I tried doing SHAPing on our training data and as our training data has around 36000 rows and 40 features, it will take almost forever to run the SHAPing, for 500 rows of data it took 5 minutes, as I will keep increasing the number of rows, it will take much more time. The below images are the values obtained for first 500 rows of SHAPing of the train data. What do you suggest we do? We go with this? or increase the rows ? Or just drop the idea of doing this? Thanks in advance !!
![image](https://user-images.githubusercontent.com/35167740/144385107-ea1cf509-93d6-4be0-a077-bb31b4e98920.png)
![image](https://user-images.githubusercontent.com/35167740/144385148-ff311d0e-04b2-4075-b38f-65c21024a28d.png)

**Comment by PhilsChan** [comment]
Hi @Ivyqiuhan I have a similar question.
For feature selection, I have tried various algorithm but it take so long (1 hour and more) to get the model ready and do hyperparameter tuning. Is it acceptable that we decide not to do any feature selection and justify our decision on the final report?
Thanks :)
